tests/v2/validation/rbac/rbac_namespace_move_test.go

```diff
@@ -0,0 +1,220 @@
+	adminProject                *management.Project
+	standardUserProjectOwner    *management.Project
+	standardUserProjectMember   *management.Project
+	standardUserProjectReadOnly *management.Project
+	steveAdminClient            *v1.Client
+	steveNonAdminClient         *v1.Client
+	namespace                   string
+	clusterName                 string
+}
+
+func (rb *NameSpaceMoveTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *NameSpaceMoveTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func (rb *NameSpaceMoveTestSuite) createProjectWithUserRole(role string) (*management.Project, error) {
+	createdProject, err := createProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+
+	switch role {
+	case roleProjectOwner:
+		rb.standardUserProjectOwner = createdProject
+		err = users.AddProjectMember(rb.client, rb.standardUserProjectOwner, rb.standardUser, role)
+
+	case roleProjectMember:
+		rb.standardUserProjectMember = createdProject
+		err = users.AddProjectMember(rb.client, rb.standardUserProjectMember, rb.standardUser, role)
+
+	case roleProjectReadOnly:
+		rb.standardUserProjectReadOnly = createdProject
+		err = users.AddProjectMember(rb.client, rb.standardUserProjectReadOnly, rb.standardUser, role)
+	}
+
+	return createdProject, err
+}
+
+func (rb *NameSpaceMoveTestSuite) createProjects() {
+	newUser, err := users.CreateUserWithRole(rb.client, users.UserConfig(), standardUser)
+	require.NoError(rb.T(), err)
+
+	rb.standardUser = newUser
+	rb.T().Logf("Created user: %v", rb.standardUser.Username)
+	rb.standardUserClient, err = rb.client.AsUser(newUser)
+	require.NoError(rb.T(), err)
+
+	rb.createProjectWithUserRole(roleProjectOwner)
+	rb.createProjectWithUserRole(roleProjectMember)
+	rb.createProjectWithUserRole(roleProjectReadOnly)
+}
+
+func (rb *NameSpaceMoveTestSuite) createNameSpace(fromProject *management.Project, role string) (*v1.SteveAPIObject, error) {
+	fromProjectName := namegen.AppendRandomString("testns-")
+	fromProjectNameSpace, err := namespaces.CreateNamespace(rb.client, fromProjectName+"-project-namespace-"+role, "{}", map[string]string{}, map[string]string{}, fromProject)
+	require.NoError(rb.T(), err)
+	return fromProjectNameSpace, err
+}
+
+func moveNameSpaceFromAProjectRoleToAProject(rb *NameSpaceMoveTestSuite, fromProject *management.Project, fromRole string, toProject *management.Project, toRole string, standardUserClient *v1.Client, isSuccessful bool) {
```

Comment:
Why is this function taking suite parameter, but some are member functions. There isn't consistency with your design at all.